script/lib/lint-less-paths.js

```diff
@@ -1,64 +1,33 @@
 'use strict'
 
-const csslint = require('csslint').CSSLint
-const expandGlobPaths = require('./expand-glob-paths')
-const LessCache = require('less-cache')
+const stylelint = require('stylelint')
 const path = require('path')
-const readFiles = require('./read-files')
+const {flatten} = require('underscore-plus')
 
 const CONFIG = require('../config')
-const LESS_CACHE_VERSION = require('less-cache/package.json').version
 
 module.exports = function () {
-  const globPathsToLint = [
-    path.join(CONFIG.repositoryRootPath, 'static/**/*.less')
-  ]
-  const lintOptions = {
-    'adjoining-classes': false,
-    'duplicate-background-images': false,
-    'box-model': false,
-    'box-sizing': false,
-    'bulletproof-font-face': false,
-    'compatible-vendor-prefixes': false,
-    'display-property-grouping': false,
-    'duplicate-properties': false,
-    'fallback-colors': false,
-    'font-sizes': false,
-    'gradients': false,
-    'ids': false,
-    'important': false,
-    'known-properties': false,
-    'order-alphabetical': false,
-    'outline-none': false,
-    'overqualified-elements': false,
-    'regex-selectors': false,
-    'qualified-headings': false,
-    'unique-headings': false,
-    'universal-selector': false,
-    'vendor-prefix': false
-  }
-  for (let rule of csslint.getRules()) {
-    if (!lintOptions.hasOwnProperty(rule.id)) lintOptions[rule.id] = true
-  }
-  const lessCache = new LessCache({
-    cacheDir: path.join(CONFIG.intermediateAppPath, 'less-compile-cache'),
-    fallbackDir: path.join(CONFIG.atomHomeDirPath, 'compile-cache', 'prebuild-less', LESS_CACHE_VERSION),
-    syncCaches: true,
-    resourcePath: CONFIG.repositoryRootPath,
-    importPaths: [
-      path.join(CONFIG.intermediateAppPath, 'static', 'variables'),
-      path.join(CONFIG.intermediateAppPath, 'static')
-    ]
-  })
-  return expandGlobPaths(globPathsToLint).then(readFiles).then((files) => {
-    const errors = []
-    for (let file of files) {
-      const css = lessCache.cssForFile(file.path, file.content)
-      const result = csslint.verify(css, lintOptions)
-      for (let message of result.messages) {
-        errors.push({path: file.path.replace(/\.less$/, '.css'), lineNumber: message.line, message: message.message, rule: message.rule.id})
-      }
-    }
-    return errors
-  })
+  return stylelint
+    .lint({
+      files: path.join(CONFIG.repositoryRootPath, 'static/**/*.less'),
+      configBasedir: __dirname,
+      configFile: path.resolve(__dirname, '..', '..', 'stylelint.config.js')
+    })
+    .then(({results}) => {
+      return flatten(
+        results.filter(_ => _.errored).map(result => {
+          const errors = result.warnings.filter(_ => _.severity === 'error')
```

Comment:
This is explicitly ignoring all non-`error`s `warnings`.

In addition the other results, including `invalidOptionWarnings`, `deprecations`, and `ignored`, are also being ignored.

Is that what is intended here?

Reply:
Yeah, this is intentional. I didn't think any of those other types of results warranted failing the build.

I'll push an update that logs these to `stdout` without going through the lint infrastructure (which doesn't handle severity levels). What do you think?

Reply:
> Yeah, this is intentional. I didn't think any of those other types of results warranted failing the build.

Excellent point, I forgot that what this plugs into has no severity levels 😆.

> I'll push an update that logs these to stdout without going through the lint infrastructure

Makes sense to me, that way they would at least be logged somewhere, and if somebody happens to catch them or spot them when looking at a failing build they can get fixed.

I just didn't want things like `deprecations` to be silently completely ignored until they turned into errors later on 😉.